src/main/java/com.company/tree/IsValidBstInConstantSpace.java

```diff
@@ -0,0 +1,34 @@
+package com.company.tree;
+
+public class IsValidBstInConstantSpace {
+    public boolean isValidBst(TreeNode root) {
+        TreeNode current = root;
+        TreeNode previous = null;
+        while (current != null) {
+            TreeNode temp = current.left;
+            if (temp != null) {
+                while (temp.right != null && temp.right != current) {
+                    temp = temp.right;
+                }
+                if (temp.right == current) {
+                    temp.right = null;
+                    if (previous.val >= current.val) {
```

Comment:
`previous` could be null. Check for it.

Reply:
it will not be null at this place

Reply:
As per the traversal 

temp.right == current
this will be true 
after we visited the left most node

Reply:
Ok, makes sense